pkg/external_repos/introspect.go

```diff
@@ -164,9 +174,62 @@ func IntrospectAll(urls *[]string, force bool) (int64, []error, []error) {
 	if err != nil {
 		errors = append(errors, err)
 	}
+
+	// Logic to handle notifications
+	sendIntrospectionNotifications(introspectSuccessUuids, introspectFailedUuids, dao)
+
 	return total, introspectionErrors, errors
 }
 
+func sendIntrospectionNotifications(successUuids []string, failedUuids []string, dao *dao.DaoRegistry) {
+	count := 0
+	wg := sync.WaitGroup{}
+
+	if len(successUuids) != 0 {
+		for i := 0; i < len(successUuids); i++ {
+			uuid := successUuids[i]
+			repos := dao.RepositoryConfig.InternalOnly_ListRepoConfigsByUUID(uuid)
+			for j := 0; j < len(repos); j++ {
+				if count > 100 { // This limits the thread count
+					wg.Wait()
+				}
+				wg.Add(1)
+				count = count + 1
+				go func(index int) {
+					notifications.SendNotification(
+						repos[index].OrgID,
+						notifications.RepositoryIntrospected,
+						[]repositories.Repositories{notifications.MapRepositoryResponse(repos[index])},
+					)
+					count = count - 1
+					wg.Done()
```

Comment:
Goroutines can't share variables from outside their scope like you're doing with `count = count - 1`. Since each parallel goroutine is simultaneously updating that variable, the result is undefined.

You might do instead is reset the counter under the `if count > 100` block. And with that, I think you would also want to call `wg.Wait()` after the for loop, to wait for whatever's left.

Reply:
Thanks for that, I also caught an error in the notificationFailure loop.
Please take a look at the logic in the follow-up commit and let me know what you think @rverdile. 

Cheers